printing/proof_diffs.ml

```diff
@@ -331,8 +317,10 @@ let goal_info goal sigma =
   in
 
   try
-    let { ty=ty; hyps=hyps; env=env } = process_goal sigma goal in
-    List.iter (build_hyp_info env sigma) (List.rev hyps);
+    let { ty=ty; env=env } = process_goal sigma goal in
+    (* compaction is usually desired [eg for better display] *)
+    let hyps = Termops.compact_named_context (Environ.named_context env) in
```

Comment:
The code has been working displaying compacted hypotheses.  What does this line add?

Reply:
This is just moving code around, the change is 100% equivalent semantically.